File: Sources/autoDarkMode/Application/SwitchMode.swift
import Foundation

/// アプリの外観切り替えモード
enum SwitchMode: String, CaseIterable, Sendable {
    /// 切り替え無効
    case off
    /// 環境光センサーによる自動切り替え
    case auto
    /// 画面輝度による手動切り替え（MAX=ライト、それ以外=ダーク）
    case manual

    var displayName: String {
        switch self {
        case .off:
            return "Off"
        case .auto:
            return "Auto"
        case .manual:
            return "Manual"
        }
    }

    var menuDescription: String {
        switch self {
        case .off:
            return "Switching disabled."
        case .auto:
            return "Automatic switching by ambient light."
        case .manual:
            return "Manual switching by display brightness."
        }
    }
}

File: src/commonMain/kotlin/com/github/gutugutu3030/autodarkmode/shared/SwitchMode.kt
package com.github.gutugutu3030.autodarkmode.shared

/**
 * 設定の切り替えモードを表します。
 *
 * @property rawValue 永続化に使う生の文字列表現です。
 * @property displayName UI に表示する名前です。
 * @property menuDescription メニューや説明文で使う簡潔な説明です。
 */
enum class SwitchMode(
    val rawValue: String,
    val displayName: String,
    val menuDescription: String,
) {
    /** 自動切り替えを無効にします。 */
    Off(
        rawValue = "off",
        displayName = "Off",
        menuDescription = "Switching disabled.",
    ),
    /** 周囲光に応じた自動切り替えを有効にします。 */
    Auto(
        rawValue = "auto",
        displayName = "Auto",
        menuDescription = "Automatic switching by ambient light.",
    ),
    /** 画面輝度の変化に応じた手動切り替えを有効にします。 */
    Manual(
        rawValue = "manual",
        displayName = "Manual",
        menuDescription = "Manual switching by display brightness.",
    );

    companion object {
        /**
         * 永続化された文字列表現からモードを復元します。
         *
         * @param raw 保存済みの生文字列です。
         * @return 一致するモード。見つからない場合は `null` を返します。
         */
        fun fromRawValue(raw: String?): SwitchMode? = entries.find { it.rawValue == raw }
    }
}